web-platform/website/src/services/AuthenticationService.tsx

```diff
@@ -53,6 +57,19 @@ export function useAuthentication() {
   return useContext(AuthenticationContext);
 }
 
+// Provides the currently logged in user for non-context caller.
+export async function getCurrentUser() {
+  const { auth } = useFirebase();
+
+  const user: FirebaseUser | null = await new Promise((resolve) => {
+    onAuthStateChanged(auth, (usr) => {
```

Comment:
Does this trigger correctly even if the user is already in the signed-in state when this is called?

Also, do we need to unsubscribe to prevent listener leak in this instance?

Reply:
(I can't figure out from the documentation so just want to make sure)

Reply:
Yes it does trigger when user is in signed in mode. The API is terrible I agree.